### PR TITLE
Feature/cirrus 143 implement exception handling middleware

### PR DESCRIPTION
**Pull request by Jakkob**

### CODE REVIEW

File: lib/core_api_ruby/error.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+module Unikey
+  # The Unikey base Error class
+  class Error < StandardError
+    attr_reader :status, :headers, :body
+    ERROR_STATUSES = 400...600
+
+    def self.retrieve_error(env)
+      klass = case env[:status].to_i
```
Comment (havok2905) [inline lib/core_api_ruby/error.rb]: Is it possible for env to be nil here? This would cause an undefined method `[]` error and raise an exception.

Comment (Jakkob) [inline lib/core_api_ruby/error.rb]: Unlikely. This is called by the object that creates `env` and, as far as I am aware, `env` is always there.

File: spec/core_api_ruby/rest/client_spec.rb [hunk 2]
```diff
@@ -28,6 +28,9 @@ module REST
         client = Client.new do |config|
           config.access_token = '574e1b42-9374-400f-b795-f5cf41d86c4e'
         end
+        client.connection = ConnectionFactory.generate_authenticated_connection(client.configuration) do |stub|
+          stub.head('/2015071001', {}) { [200, {}, nil] }
```
Comment (jlorich) [inline spec/core_api_ruby/rest/client_spec.rb]: even though this is just a test, we should have the base url come from a config somewhere

Comment (jlorich) [inline spec/core_api_ruby/rest/client_spec.rb]: there's a number of places it's referenced and the tests could potentially break if we change api versions

File: bin/run_checks [hunk 3]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env bash
 
-clint --lint .
+clint --lint lib spec
```
Comment (jlorich) [inline bin/run_checks]: should add this to .travis.yml too

File: spec/core_api_ruby/rest/client_spec.rb (hunk 2 above)
Comment (Jakkob) [inline spec/core_api_ruby/rest/client_spec.rb]: There already is a static constant. It is inserted into the connection earlier in the call stack.
I made a `BASE_API_VERSION` constant as well. It seems that this only is an issue when you're making a head request.
Fixed.

**Pull request merged**